compiler/rustc_middle/src/ty/diagnostics.rs

```diff
@@ -426,6 +426,8 @@ impl<'tcx> TypeVisitor<'tcx> for IsSuggestableVisitor<'tcx> {
 
     fn visit_ty(&mut self, t: Ty<'tcx>) -> ControlFlow<Self::BreakTy> {
         match t.kind() {
+            Infer(InferTy::TyVar(_)) => {}
```

Comment:
I doubt that this is the correct approach.

Reply:
This is _almost_ the right approach -- instead, can you make it a boolean? Like `infer_suggestable`? And edit all of the `is_suggestable` calls to be false except for the one you added?